### ISSUE TITLE
Multiple small issues

### ISSUE THREAD

**Issue by ocornu** [ponewheel/android-ponewheel#35]
Hi,
I'm running the Google Play version (2.3), so I'm afraid what I'm gonna say next has long been fixed. That's why I'll bundle them in a single ticket here. By the way, I'd love some documentation on how to build this —I've never built an android app before!

1. Selecting the "Metrics Units" checkbox does not work over app restart (option is selected ok, but metrics units won't show up until you unselect/reselect it in the settings.

**Comment by ebabel** [comment]
@ocornu Thanks for this great feedback and suggestions. Will fix asap.

**Comment by ebabel** [comment]
Fixed metric units problem in #36

**Comment by ebabel** [comment]
@kwatkins do we know if the device is plugged in to the charger?

**Comment by ebabel** [comment]
@ocornu What sound would you want to hear for the speed alert?

**Comment by ocornu** [comment]
> do we know if the device is plugged in to the charger?

Seems so:
![image](https://user-images.githubusercontent.com/54847/33874602-a5ad8dc8-df1f-11e7-9246-115b591df2b7.png)

> What sound would you want to hear for the speed alert?

Well, any sound would do, but: good point. :)
Mmh... How about a cop/ambulance siren sound?  Something that suggests breaking the speed limit, or getting closer to riding back in an ambulance... :D

**Comment by ocornu** [comment]
@ebabel lol
Perhaps a shorter version of http://soundbible.com/581-Police-Siren-3.html ?
I like http://soundbible.com/45-Police-Siren.html better, but its licensed for personal use only. :/

**Comment by ocornu** [comment]
For the record; the max speed alert idea comes from OneDan+ on the OW forum: http://community.onewheel.com/topic/7477/customizable-top-speed-warning-please

**Comment by ebabel** [comment]
Thanks for the link, there is a lot of interest in this feature!

**Comment by kwatkins** [comment]
@ocornu great ideas! 

re: getting the charging state @ebabel I just sadly confirmed that flag doesn't set when the OW is charging (deviceStatus.charging). It's always false. You could keep state on the amps added or the battery %, that should work.

**Comment by wmaciel** [comment]
The official app knows when it is charging, there's gotta be a way to know.

I think having a watch on the battery percentage might not be enough as if you start going downhill and end up "charging" the battery past the set limit you might confuse some people.

Maybe if the speed is also zero we could fire the alarm, but thats a bit contrived, I think.

**Comment by ebabel** [comment]
hmm, for now I'll just always play the sound when the checkbox is checked and the percent is over the user's requested percent. They can turn it off when they're done charging and back on again when they start.
If you guys think of something better let me know.

**Comment by ocornu** [comment]
> The official app knows when it is charging, there's gotta be a way to know.

Yeah, bit 15 of the status register (the infamous "status 8192") isn't it?

**Comment by ocornu** [comment]
It seems official app is giving status 8196 when full. Is bit 2 a "battery full" flag?

**Comment by ebabel** [comment]
> Optional speed alert that would play a sound when you go beyond a set speed. Additional safety against high-speed nosedives.
Optional charge alert that would play a sound when you reach a certain level. Would help you stop recharging before 100% when starting from a hilltop.

Fixed in #38 

still need:

> When disconnected/reconnected (board power cycle for example), the state of the pads is not updated until you step back on it

**Comment by ocornu** [comment]
Awesome, thanks!
You guys are super reactive, kudos!
